Dog Page - View a singular dog

UZ9 opened the pull request:
## PR Title
Issue Number(s): #5

### Addition of /dogs
Although the primary features listed in #4 have not been added to this commit (e.g. responsiveness of table), a rudimentary interface for viewing each Dog has been added.

### Single Dog Page

The single dog page includes relevant information pulled from the Dog schema. As certain elements are unconfirmed, some properties are left as placeholders.

### Relevant Photos
![image](https://github.com/GTBitsOfGood/canine-assistants/assets/36551149/0f0936cf-3cbc-4067-8a77-69f09700f334)

![image](https://github.com/GTBitsOfGood/canine-assistants/assets/36551149/35cc8820-2158-4402-ad1c-0784524a378c)

![image](https://github.com/GTBitsOfGood/canine-assistants/assets/36551149/2cefc22e-9484-4f8e-992f-7a9d48dcb892)

What does this PR change and why?
### Checklist
- [x] Requirements have been implemented
- [x] Acceptance criteria is met
- [x] Database schema docs have been updated or are not necessary
- [x] Code follows design and style guidelines
- [x] Commits follow guidelines (concise, squashed, etc)
- [x] Relevant reviewers (EM) have been assigned to this PR

### How To Test
Navigate to /dogs to view the main table layout. For viewing a specific Dog, click on the dog's name to view the singular page.

UZ9 committed:
Add individual dog page + components

UZ9 committed:
Merge branch 'dev' of https://github.com/GTBitsOfGood/canine-assistants into dev

UZ9 committed:
Merge branch 'dev' of https://github.com/GTBitsOfGood/canine-assistants into dev

UZ9 committed:
Merge branch 'dev' of https://github.com/GTBitsOfGood/canine-assistants into dev

UZ9 committed:
:sparkles: Implement single page Dog view + More

- Created re-usable Card component, handy for elements requiring a label in the top left and a drop shadow
- Created modular and extendable Table component, allowing data to easily be styled (e.g. banded columns) and automatic date conversion. Additionally, each element of the table can optionally have a custom render function for later features, e.g. chips
- Created barebones /dogs page to easily navigate to each individual dog, where the name of each animal can be clicked. Note that this is NOT an entire completion of #4.
-  Abstracted main dogs table to its own component with a temporary search feature
- Created a PetmateCard component dedicated to rendering the relationships of each dog (and can be easily added onto in the future)
- Modified TabSection to be responsive, as the additional padding was causing the container to break on mobile sizes
- Refactored individual dog page to be fully responsive no matter the size
- Added mock data to the consts file
- Added complete JSDocs to all created components

UZ9 committed:
:hammer: Refactor col html to avoid redundant code

afazio1 commented on server/db/dbConnect.js:
```diff
@@ -33,6 +33,8 @@ async function dbConnect() {
     cached.promise = mongoose
       .connect(`${DB_URL}${DB_NAME}?retryWrites=true&w=majority`, opts)
       .then((mongoose) => {
+        mongoose.set('debug', true)
```
Could we make this conditional based on if the environment is in dev vs prod? I'm thinking this may required an additional env variable.

UZ9 replied:
No additional env variable needed thanks to NODE_ENV!

afazio1 reviewed (changes requested):
Requesting a minor change for db settings code. Everything else looks great! FE code is setup in a way that's robust/maintainable and UI looks awesome. Great work!

UZ9 committed:
:construction_worker: Only enable mongoose debug in dev

Mongoose debug property is now dependent on the NODE_ENV setting.

By default NODE_ENV is set to 'development'. For all deployed instances, NODE_ENV will be set to 'production'.

afazio1 reviewed (approved):
Good 2 go 🎉
